### ISSUE TITLE
Barcodescanner is not working on iOS in a modal page

### ISSUE THREAD

**Issue by bnbs** [EddyVerbruggen/nativescript-barcodescanner#133]
When trying to open the barcodescanner in a modal page the barcodescanner is not being showed and this only happens on iOS. The demo below works on android but didn't work on iOS.
[demo.zip](https://github.com/EddyVerbruggen/nativescript-barcodescanner/files/1665437/demo.zip)

**Comment by isaacfi** [comment]
I have the same issue, any news about this?

**Comment by t3hmrman** [comment]
Hey I just ran into this, and in case this is still an issue, it's [a limitation of iOS](https://github.com/NativeScript/NativeScript/issues/423) -- by initializing the scanner and trying to call `.scan`, it will attempt to open a *new* modal -- if you try and do this from another modal (for example, one that let you choose *how* to retrieve the QR, which is probably a common scheme), you will get an error like this:

```
Warning: Attempt to present <QRCodeReaderViewController: 0x14fe80140> on <UINavigationControllerImpl: 0x15017b800> whose view is not in the window hierarchy!
```

For me, I fixed this by calling `this.$modal.close()` inside the modal I had open

**Comment by EddyVerbruggen** [comment]
Yep, you’re right. 

Alternatively, embed the scanner UI in the model.

**Comment by EddyVerbruggen** [comment]
Found a fix eventually - avail in 3.0.2.

**Comment by t3hmrman** [comment]
Thanks for finding a fix!
